src/libs/Animals/TButterflies.cpp

```diff
@@ -9,6 +9,7 @@
 //--------------------------------------------------------------------
 TButterflies::TButterflies() : enabled(false), yDefineTime(0)
 {
+    butterflies.reserve(BUTTERFLY_COUNT);
```

Comment:
vector::reserve only allocates capacity, it does not change the size of the vector. I don't see where the size of the vector is changed via vector::push_back or vector::resize. Are you sure your vector does not end up empty?